### PR TITLE
BA-23-Implement-page-object-for-cart - first version to review

### PR DESCRIPTION
**Pull request by claudiawozniak**
## Description:
This pull request introduces cartPage package, which contains several Java classes related to different functionalities on the cart page of a web application. 

!Important! without product details table (it will be delivered with Ba 29 implementation of table with product details cart page object)

## Changes:

 `CartPage:`
- This class represents the cart page in the web application.
- It utilizes the org.openqa.selenium.By class and the WebDriver class.
- It contains locators for elements on the page, such as the `updateCartButton`, `checkoutButtonLocator`, `continueShoppingButtonLocator` and `finalCheckoutButtonLocator`. 
- It also contains methods for interacting with these elements, such as clicking the update cart button (`clickOnUpdateButton`), continue shopping (`clickOnContinueShoppingButton`) or final checkout button (`clickOnFindalCheckoutButton`).

##
`ApplyCouponCodeTable:`
- This class represents the area on the cart page used for entering a coupon code.
- It utilizes the org.openqa.selenium.By class for locating elements on the page using XPath.
- It contains constants representing XPath paths to elements on the page, such as the `couponLocator` input field and `applyCouponButtonLocator`.
- It also contains methods like `enterCouponCode` and`clickOnApplyCouponButton`.

##
`EstimateShippingAndTaxesTable:`
- This class represents the area on the cart page used for estimating shipping and taxes.
- Similar to the previous class, it utilizes the org.openqa.selenium.By class.
- It contains constants representing XPath paths to elements on the page, such as `countryDropdownLocator`, `stateDropdownLocator`, `zipPostCodeLocator` input field, `estimateButtonLocator` and `shipmentsDropdownLocator`.
- It also contains methods like `enterZipPostCode` and `clickOnEstimateButton`.

##
`TotalsTable:`
- This class represents the table displaying various totals on the cart page, such as subtotal, flat shipping rate, total.
- It utilizes the org.openqa.selenium.By class.
- It contains constants representing XPath paths to elements in the table on the page, such as `subTotalLocator`, `flatShippingRateLocator`, `totalLocator`.
- It also contains methods like `getSubTotalValue`, `getFlatShippingRateValue` and `getTotalValue`.

### CODE REVIEW

File: src/main/java/cartPage/TotalsTable.java
Comment (Fishgame1) [inline src/main/java/cartPage/TotalsTable.java:12]: It will be good to have some validation methods here?:)

File: src/main/java/cartPage/EstimateShippingAndTaxesTable.java
Comment (Fishgame1) [inline src/main/java/cartPage/EstimateShippingAndTaxesTable.java:9]: It will be good to have some validation methods here?:)

File: src/main/java/cartPage/ApplyCouponCodeTable.java
Comment (Fishgame1) [inline src/main/java/cartPage/ApplyCouponCodeTable.java:8]: It will be good to have some validation methods here?:)

File: src/main/java/cartPage/CartPage.java
Comment (Fishgame1) [inline src/main/java/cartPage/CartPage.java:8]: I think this page should have other classes from this package as parameters?

**Review by Fishgame1 — CHANGES_REQUESTED** [review]

File: src/main/java/cartPage/ApplyCouponCodeTable.java
Comment (Fishgame1) [inline src/main/java/cartPage/ApplyCouponCodeTable.java:7]: This might be potentially shortened.

File: src/main/java/cartPage/TotalsTable.java
Comment (claudiawozniak) [reply src/main/java/cartPage/TotalsTable.java:12]: Of course but it will be useful in the future, because it is not needed for our current epic. But I can add a few methods :)

File: src/main/java/cartPage/EstimateShippingAndTaxesTable.java
Comment (claudiawozniak) [reply src/main/java/cartPage/EstimateShippingAndTaxesTable.java:9]: The same as above. I will add some methods there

File: src/main/java/cartPage/ApplyCouponCodeTable.java
Comment (claudiawozniak) [reply src/main/java/cartPage/ApplyCouponCodeTable.java:8]: The same as above. I will add some methods there

File: src/main/java/cartPage/ApplyCouponCodeTable.java
Comment (claudiawozniak) [reply src/main/java/cartPage/ApplyCouponCodeTable.java:7]: Ok. Done

File: src/main/java/cartPage/CartPage.java
Comment (claudiawozniak) [reply src/main/java/cartPage/CartPage.java:8]: Ok, done.

File: src/main/java/pages/cartPage/TotalsTable.java
Comment (Fishgame1) [inline src/main/java/pages/cartPage/TotalsTable.java:30]: We don't need that in page object.

File: src/main/java/pages/cartPage/TotalsTable.java
Comment (Fishgame1) [inline src/main/java/pages/cartPage/TotalsTable.java:22]: remove empty line

File: src/main/java/pages/cartPage/EstimateShippingAndTaxesTable.java
Comment (Fishgame1) [inline src/main/java/pages/cartPage/EstimateShippingAndTaxesTable.java:26]: remove empty line.

File: src/main/java/pages/cartPage/CartPage.java
Comment (Fishgame1) [inline src/main/java/pages/cartPage/CartPage.java:35]: empty lines.

File: src/main/java/pages/cartPage/CartPage.java
Comment (Fishgame1) [inline src/main/java/pages/cartPage/CartPage.java:9]: Why this parameter?

File: src/main/java/pages/cartPage/ApplyCouponCodeTable.java
Comment (Fishgame1) [inline src/main/java/pages/cartPage/ApplyCouponCodeTable.java:9]: Why we need this parameter?

File: src/main/java/pages/cartPage/ApplyCouponCodeTable.java
Comment (Fishgame1) [inline src/main/java/pages/cartPage/ApplyCouponCodeTable.java:10]: This should be parameter of enterCouponCode() method

File: src/main/java/cartPage/CartPage.java
Comment (Fishgame1) [reply src/main/java/cartPage/CartPage.java:12]: maybe you can use that:
.totals > a:nth-of-type(1)

File: src/main/java/cartPage/EstimateShippingAndTaxesTable.java
Comment (Fishgame1) [inline src/main/java/cartPage/EstimateShippingAndTaxesTable.java:3]: I think you forgot to remove old pages from the PR:)

**Review by Fishgame1 — CHANGES_REQUESTED** [review]

File: src/main/java/cartPage/CartPage.java
Comment (claudiawozniak) [reply src/main/java/cartPage/CartPage.java:12]: Done

File: src/main/java/pages/cartPage/ApplyCouponCodeTable.java
Comment (claudiawozniak) [reply src/main/java/pages/cartPage/ApplyCouponCodeTable.java:9]: We dont. Removed

File: .gitignore
Comment (claudiawozniak) [reply .gitignore:65]: Ok. Added again.

File: src/main/resources/config.example
Comment (claudiawozniak) [reply src/main/resources/config.example:5]: Yes, because guys made the same changes in theirs, so I did it to before I will solve the merge problems

File: src/test/java/temp/tests/TestCartPage.java
Comment (claudiawozniak) [reply src/test/java/temp/tests/TestCartPage.java:18]: Ok. Moved

File: .gitignore
Comment (Fishgame1) [inline .gitignore:64]: it should be temp/

Comment (claudiawozniak) [reply .gitignore:64]: Ah. U r right :)

**Review by jakub-krzysiek — APPROVED** [review]
@claudiawozniak Approved :)

File: src/main/java/steps/MainPageSteps.java
Comment (Fishgame1) [inline src/main/java/steps/MainPageSteps.java:48]: Oh...sorry those should be in the CartPageSteps class not here:P

Comment (claudiawozniak) [reply src/main/java/steps/MainPageSteps.java:48]: Done! Im waiting for approval! :D

**Review by Fishgame1 — APPROVED** [review]

**Pull request merged**
